sensor/common/networkflow/manager/manager_impl.go

```diff
@@ -235,27 +250,53 @@ func (e *containerEndpoint) String() string {
 	return fmt.Sprintf("%s %s: %s", e.containerID, e.processKey, e.endpoint)
 }
 
+type Option func(*networkFlowManager)
+
+// WithEnrichTicker overrides the default enrichment ticker
+func WithEnrichTicker(ticker <-chan time.Time) Option {
+	return func(manager *networkFlowManager) {
+		if ticker != nil {
+			manager.enricherTickerC = ticker
+		}
+	}
+}
+
+// WithPurger binds the manager with the purger.
+// Note that `networkFlowManager` being unexported prevents us from implementing it the other way round:
+// `NewPurger(..., ForManager(manager))`.
+func WithPurger(purger *NetworkFlowPurger) Option {
+	return func(manager *networkFlowManager) {
+		if purger != nil {
+			purger.manager = manager
+		}
+	}
+}
+
 // NewManager creates a new instance of network flow manager
 func NewManager(
 	clusterEntities EntityStore,
 	externalSrcs externalsrcs.Store,
 	policyDetector detector.Detector,
 	pubSub *internalmessage.MessageSubscriber,
+	opts ...Option,
 ) Manager {
-	enricherTicker := time.NewTicker(tickerTime)
+	enricherTicker := time.NewTicker(enricherCycle)
+
 	mgr := &networkFlowManager{
 		connectionsByHost: make(map[string]*hostConnections),
 		clusterEntities:   clusterEntities,
 		publicIPs:         newPublicIPsManager(),
 		externalSrcs:      externalSrcs,
 		policyDetector:    policyDetector,
 		enricherTicker:    enricherTicker,
+		enricherTickerC:   enricherTicker.C,
 		initialSync:       &atomic.Bool{},
-		activeConnections: make(map[connection]*networkConnIndicator),
-		activeEndpoints:   make(map[containerEndpoint]*containerEndpointIndicator),
+		activeConnections: make(map[connection]*networkConnIndicatorWithAge),
+		activeEndpoints:   make(map[containerEndpoint]*containerEndpointIndicatorWithAge),
 		stopper:           concurrency.NewStopper(),
 		pubSub:            pubSub,
 	}
+	mgr.purger = NewNetworkFlowPurger(clusterEntities, env.EnrichmentPurgerTickerMaxAge.DurationSetting(), mgr)
```

Comment:
Maybe we should make sure `MaxAge` is valid (e.g. MaxAge > enricherCycle).

Reply:
oh, that is a very good idea!

Reply:
In the case where it is not valid, would it make sense to extend `MaxAge` to a higher value than `enricherCycle` and log the configuration error ?

Reply:
here it is [fdbfa3c](https://github.com/stackrox/stackrox/pull/14538/commits/fdbfa3cb6c594df1fb3312504cd6547f6697f252)